test/functional/core/path_spec.lua

```diff
@@ -91,4 +91,80 @@ describe('file search', function()
     feed('gf')
     eq('filename_with_unicode_ααα', eval('expand("%:t")'))
   end)
+
+  it('matches Windows drive-letter filepaths (without ":" in &isfname)', function()
+    local os_win = is_os('win')
+
+    insert([[c:/d:/foo/bar.txt]])
+    eq([[c:/d:/foo/bar.txt]], eval('expand("<cfile>")'))
```

Comment:
This doesn't seem right. If ":" is not in 'isfname' why is the 2nd drive letter allowed? I would expect the result to be `d:/foo/bar.txt`

Not a blocker for this PR I guess.

Reply:
My guess is that this happens because of the `path_is_url` implementation. Since just a `:\` characterizes the path as a valid URL then Neovim now considers the following `:` as a valid character because it is used for port specification.

If this becomes a big deal then we might need to change `path_is_url` to implement a URL that follows the [RFC 7230](https://datatracker.ietf.org/doc/html/rfc7230#section-2.7).

Reply:
`path_is_url` should definitely be improved, an easy improvement would be to treat single-char schemes as drive letters (i.e. "not a url"). That's mentioned at https://url.spec.whatwg.org/#start-with-a-windows-drive-letter